presto-main/src/main/java/com/facebook/presto/sessionpropertyproviders/NativeWorkerSessionPropertyProvider.java

```diff
@@ -311,6 +313,20 @@ public NativeWorkerSessionPropertyProvider(FeaturesConfig featuresConfig)
                         "Minimum amount of data processed by all the logical table partitions " +
                                 "to trigger skewed partition rebalancing by scale writer exchange.",
                         256L << 20,
+                        !nativeExecution),
+                booleanProperty(
+                        NATIVE_TABLE_SCAN_SCALED_PROCESSING_ENABLED,
+                        "If set to true, enables scaling the table scan concurrency on each worker.",
+                        false,
+                        !nativeExecution),
+                doubleProperty(
+                        NATIVE_TABLE_SCAN_SCALE_UP_MEMORY_USAGE_RATIO,
+                        "The query memory usage ratio used by scan controller to decide if it can " +
```

Comment:
would be good to add validation so we fail fast if an invalid property is sent.  you can look at https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java#L1026-L1034 for an example of how to do that.  

Does cpp have any validation options for session properties?

Reply:
I don't see any Validation File specific to cpp. I do see a Test file, where I've added these session properties. 

From the file you [shared](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java#L330) "Native execution related session properties are temporarily put here." Should I add these native configurations there as well? Does it could as execution-related?

Reply:
no, where you have them now is good. I think the ones in that file are things that the coordinator needs to know about for planning.  The configs you added the coordinator doesn't need to use at all. it just needs to know about them so it can pass them through to the worker.